projects/js-packages/shared-extension-utils/src/get-block-icon-from-metadata.native.js

```diff
@@ -0,0 +1,33 @@
+/* eslint-disable jsdoc/no-undefined-types */
+import { SvgXml } from 'react-native-svg';
+import getIconColor from './get-icon-color';
+
+/**
+ * Generate an icon as a React component from the SVG markup defined in a block.json metadata file.
+ * This prevents us from duplicating the markup in various places.
+ *
+ * Note: using an `img` tag and passing the SVG markup as a data URI doesn't allow us to
+ * dynamically set the icon color later on.
+ *
+ * @param {object} metadata - Block.json content
+ * @returns {JSX.Element} Icon component
+ */
+export function getBlockIconComponent( metadata ) {
+	return <SvgXml xml={ metadata.icon } />;
```

Comment:
What happens if the block has no `metadata.icon`, or has it set to a non-svg like instagram-gallery currently does?
https://github.com/Automattic/jetpack/blob/6960749e172aba929224b545fcb6461ed2423117/projects/plugins/jetpack/extensions/blocks/instagram-gallery/block.json#L11

Reply:
@anomiex, good point! @fluiddot has suggested another approach via Slack that I'm planning to follow instead of this. I'll work on this and set as ready for review again when it's ready. (Just juggling a few tasks today, sorry for not being super responsive!)

Reply:
Noting that, to address this, in https://github.com/WordPress/gutenberg/pull/55742, a change has been made directly to [`components/block-icon/index.native.js`](https://github.com/WordPress/gutenberg/pull/55742/files#diff-ba3bb0b5d33390dd479b7b3eff6a656890a89fbcf11c7fd84e073968447a0bef) so that the parsing is handled on the Gutenberg side.

Reply:
Following the discovery that the icon can be passed to components other than `<BlockIcon/>`, I reverted the aforementioned change, so we're back to parsing the icon in `get-block-icon-from-metadata.native.js`. 

To account for cases where there may not be a block icon, or it may not be an svg string, I made some changes in https://github.com/Automattic/jetpack/pull/33890/commits/1a127671742c8a8729d6180702bf046340223611. I believe these changes bring us in alignment to how the logic works in the web file.

I used the `instagram` example to test that this works locally:

![Screenshot_20231106_172034](https://github.com/Automattic/jetpack/assets/2998162/6f728318-3fd3-42df-9beb-39ff93ac910b)